### PR TITLE
[MIBM-120][JR] /enter-eori-number

### PR DESCRIPTION
**Pull request by jordanrowe**

### CODE REVIEW

File: app/uk/gov/hmrc/merchandiseinbaggagefrontend/views/EoriNumberView.scala.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+layout: Layout,
+formHelper: FormWithCSRF,
+errorSummary: components.errorSummary,
+text: components.inputText,
+button: components.button
+)
+
+@(form: Form[_], isAgent: YesNo)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
+
+@conditionalKey = @{if(YesNo.to(isAgent)) "agent" else "trader"}
```
Comment (PaulHodgson) [inline app/uk/gov/hmrc/merchandiseinbaggagefrontend/views/EoriNumberView.scala.html:32]: Maybe `agentOrTrader`?

File: test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/EoriNumberControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+  }
+
+  "onSubmit" must {
+    val url = routes.EoriNumberController.onSubmit().url
+    val postRequest = buildPost(url, sessionId)
+
+    behave like anEndpointRequiringASessionIdAndLinkedDeclarationJourneyToUpdate(controller, url)
+
+    "Redirect to /traveller-details" when {
```
Comment (PaulHodgson) [inline test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/EoriNumberControllerSpec.scala:90]: Not sure we need this test given the selenium spec. Does it cover anything different?

File: test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/EoriNumberControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney)
+
+        val result = controller.onPageLoad()(getRequest)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).get mustBe routes.InvalidRequestController.onPageLoad().url
+      }
+    }
+
+    "return OK and render the view" when {
```
Comment (PaulHodgson) [inline test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/EoriNumberControllerSpec.scala:48]: do we need all 3 of these and the selenium spec?

File: app/uk/gov/hmrc/merchandiseinbaggagefrontend/views/EoriNumberView.scala.html (hunk 1 above)
Comment (jordanrowe) [reply app/uk/gov/hmrc/merchandiseinbaggagefrontend/views/EoriNumberView.scala.html]: pushed

File: test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/EoriNumberControllerSpec.scala (hunk 2 above)
Comment (jordanrowe) [reply test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/EoriNumberControllerSpec.scala:90]: Better safe than sorry. I feel like if I hadn't included both you would have asked for both 🙃

Comment (PaulHodgson) [reply test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/EoriNumberControllerSpec.scala:90]: Bad Paul!

I removed some controller tests the other day. I'd say only add what we need. 

Up to you.

**Pull request merged**